packages/stainless/src/stl.ts

```diff
@@ -190,74 +256,146 @@ export class StlError extends Error {
   }
 }
 
+/** When thrown in an endpoint handler, responds with HTTP status code 400. */
 export class BadRequestError extends StlError {
+  /**
+   * @param response  optional data included in the body of the response JSON.
+   */
   constructor(response?: Record<string, any>) {
     super(400, { error: "bad request", ...response });
   }
 }
+
+/** When thrown in an endpoint handler, responds with HTTP status code 401. */
 export class UnauthorizedError extends StlError {
+  /**
+   * @param response  optional data included in the body of the response JSON.
+   */
   constructor(response?: Record<string, any>) {
     super(401, { error: "unauthorized", ...response });
   }
 }
+
+/** When thrown in an endpoint handler, responds with HTTP status code 403. */
 export class ForbiddenError extends StlError {
+  /**
+   * @param response  optional data included in the body of the response JSON.
+   */
   constructor(response?: Record<string, any>) {
     super(403, { error: "forbidden", ...response });
   }
 }
 
+/** When thrown in an endpoint handler, responds with HTTP status code 404. */
 export class NotFoundError extends StlError {
+  /**
+   * @param response  optional data included in the body of the response JSON.
+   */
   constructor(response?: Record<string, any>) {
     super(404, { error: "not found", ...response });
   }
 }
 
 type AnyStatics = Record<string, any>; // TODO?
 
+/**
+ * The type of a `stainless` plugin.
+ */
 export type StainlessPlugin<
   Statics extends AnyStatics | undefined = undefined
 > = {
+  /**
+   * Optionally provide data to every endpoint handler.
+   * This will be available under {@link StlContext.plugins},
+   * under the field with the user's name for the plugin.
+   */
   statics?: Statics;
```

Comment:
do we actually want to keep this at all? @jedwards1211

Reply:
yeah probably not